File: sample/src/main/java/com/lyc/yuchuan_downloader/Async.java
package com.lyc.yuchuan_downloader;

import android.os.Handler;
import android.os.Looper;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

/**
 * Created by Liu Yuchuan on 2019/5/18.
 */
public class Async {
    static final Executor cache = Executors.newCachedThreadPool();
    private static final Handler HANDLER = new Handler(Looper.getMainLooper());
    static final Executor main = new Executor() {
        @Override
        public void execute(Runnable command) {
            HANDLER.post(command);
        }
    };

    private Async() {
    }
}

File: sample/src/main/java/com/lyc/yuchuan_downloader/Async.kt
package com.lyc.yuchuan_downloader

import android.os.Handler
import android.os.Looper

import java.util.concurrent.Executor
import java.util.concurrent.Executors

/**
 * Created by Liu Yuchuan on 2019/5/18.
 */
object Async {
    internal val cache: Executor = Executors.newCachedThreadPool()
    private val HANDLER = Handler(Looper.getMainLooper())
    internal val main: Executor = Executor { command -> HANDLER.post(command) }
}
